### PR TITLE
ImageViewer - limit zoom by the window/screen size

### PR DESCRIPTION
**Pull request by igordmn**
- add a `zoom` field, which is the same across different screen/window sizes

- The `scale` field is not a base state now, it is derived from the current zoom and the current screen/window size.

- drag amount is still independent of scale/zoom (if we drag by 5 pixels, the image moves by 5 pixels)

- offset is still limited by the area and the current scale

https://user-images.githubusercontent.com/5963351/230199995-f7d93a80-01b3-4cab-aeaa-f84bb2dbaa4f.mp4

### CODE REVIEW

**Comment by igordmn** [comment]
cc @dima-avdeev-jb

**Review by dima-avdeev-jb — APPROVED** [review]
Nice!
Tested, it works on all platforms.

File: examples/imageviewer/shared/src/commonMain/kotlin/example/imageviewer/view/ScalableImage.common.kt [hunk 1]
```diff
@@ -55,22 +44,22 @@ internal fun ScalableImage(scalableState: ScalableState, image: ImageBitmap, mod
...
-                    scalableState.addScale(zoom, centroid - areaCenter)
+                    scalableState.addZoom(zoom, centroid - areaCenter)
                 }
                 .pointerInput(Unit) {
                     detectTapGestures(onDoubleTap = { position ->
-                        scalableState.setScale(
-                            if (scalableState.scale > 2.0) {
-                                scalableState.scaleLimits.start
+                        scalableState.setZoom(
+                            if (scalableState.zoom > 1.5f) {
```
Comment (MatkovIvan) [inline examples/imageviewer/shared/src/commonMain/kotlin/example/imageviewer/view/ScalableImage.common.kt:59]: It's not super clear what exactly this check do and why `1.5`?

Comment (igordmn) [reply examples/imageviewer/shared/src/commonMain/kotlin/example/imageviewer/view/ScalableImage.common.kt]: Added a comment and extracted a constant

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**